docs/modules/ROOT/pages/accounts.adoc

```diff
@@ -120,6 +71,26 @@ trait ISRC5 {
 Even though these interfaces are not enforced by the protocol, it's recommended to implement them for enabling
 interoperability with the ecosystem.
 
+
+== Protocol-level methods
+
+In this section we will describe the methods that the protocol uses for abstracting the accounts. The first two
+are required for enabling accounts to be used for executing transactions. The rest are optional:
+
+1. `\\__validate__` verifies the validity of the transaction to be executed. This is usually used to validate signatures,
+but following the account abstraction design, the entrypoint implementation can be customized to feature any
+validation mechanism.
+
+2. `\\__execute__` executes the transaction if the validation is successful.
+
+3. `\\__validate_declare__` optional entrypoint similar to `\\__validate__` but for transactions
+meant to declare other contracts.
+
+4. `\\__validate_deploy__` optional entrypoint similar to `\\__validate__` but meant for {counterfactual}.
+
+NOTE: Even though these entrypoints _can_ be called directly at the contract level, they are not designed for that,
+but to be called by the Starknet OS in the transaction execution flow.
```

Comment:
i'm not sure how i feel about this note, it feels like a warning. although it's true they were "not designed for that", i kind of look forward experimentation. for example, contracts calling an account's `__execute__`

Reply:
Are you proposing to remove it? My intention was to mention that they are meant to be called by the protocol, not that you can't use them in experimentation.

Reply:
the "not designed for" sounds feels like a warning/deterrent to me. I'd probably change the wording more like 

> NOTE: Although these entrypoints are available to the protocol for its regular transaction flow, they can also be called like any other method."